query-engine/core/src/query_graph_builder/extractors/query_arguments.rs

```diff
@@ -307,27 +320,16 @@ fn extract_cursor(value: ParsedInputValue, model: &ModelRef) -> QueryGraphBuilde
     Ok(Some(SelectionResult::new(pairs)))
 }
 
-fn extract_cursor_field(
-    field: ScalarFieldRef,
-    input_value: ParsedInputValue,
-) -> QueryGraphBuilderResult<Vec<(ScalarFieldRef, PrismaValue)>> {
-    let value = input_value.try_into()?;
-    Ok(vec![(field, value)])
-}
-
-fn extract_compound_cursor_field(
-    fields: Vec<ScalarFieldRef>,
-    input_value: ParsedInputValue,
-) -> QueryGraphBuilderResult<Vec<(ScalarFieldRef, PrismaValue)>> {
-    let mut map: ParsedInputMap = input_value.try_into()?;
-    let mut pairs = vec![];
+fn extract_from_input_map(path: &[String], map: &mut ParsedInputMap) -> PrismaValue {
+    let name = path.first().expect("I was expecting a path to unwrap");
```

Comment:
Is there any chance we could hit this `expect`?

Reply:
Not really, this could be replaced with an unwrap. Though, using expect with a message at least gives somebody a context about what is a case that never happens (contrary to just read unwrap).